platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+
+**Description**: This class provides access to each group (G₁, G₂) for a specific Pairing and the FiniteField associated with the curve.
+It provides an initialization method to load all necessary dependencies for the library to work. Callers are requested to invoke that method before start using the library.
+
+**Link**:  [BilinearPairing.java](pairings-api%2FBilinearPairing.java)
+
+###### `Field`
+**Description**: This is a factory interface responsible for creating FieldElement, which are scalars belonging to the field represented by this instance.
+
+**Link**: [Field.java](pairings-api%2FField.java)
+
+###### `FieldElement`
+
+**Description**: An interface representing an element within a field, providing methods for basic arithmetic operations.
+
+**Link**: [FieldElement.java](pairings-api%2FFieldElement.java)
+
+###### `Group`
+
+**Description**: This class provides methods to obtain elements belonging to the group represented by the instance.
+
+**Link**: [Group.java](pairings-api%2FGroup.java)
+
+###### `GroupElement`
+
+**Description**: An interface representing an element within a group, providing methods for basic group operations.
+
+**Link**: [GroupElement.java](pairings-api%2FGroupElement.java)
+
+###### `PairingResult`
+
+**Description**: An interface representing the result of a pairing operation, with methods to compare it to other group elements.
+
+**Link**: [PairingResult.java](pairings-api%2FPairingResult.java)
+
+
+#### Swirlds Crypto Pairings Signature Library
+##### Overview
+This module provides cryptography primitives to create EC PublicKeys, EC PrivateKeys, and Signatures.
+
+##### Public API
+###### `SignatureSchema`
+**Link**: [SignatureSchema.java](signature-lib%2FSignatureSchema.java)
+
+**Description**: A pairings signature scheme can be implemented with different types of curves and group assignment configurations.
+For example, two different configurations might consist of a BLS_12_381 curve using G1 of the pairing to generate public key elements or G2 for the same purpose.
+###### `PairingPrivateKey`
+**Link**: [PairingPrivateKey.java](signature-lib%2FPairingPrivateKey.java)
+
+**Description**: A private key generated using the pairings API
+
+###### `PairingPublicKey`
+**Link**: [PairingPublicKey.java](signature-lib%2FPairingPublicKey.java)
+
+**Description**: A public key generated using the pairings API
+###### `PairingSignature`
+**Link**: [PairingSignature.java](signature-lib%2FPairingSignature.java)
+
+**Description**: A signature generated with the private key that can be verified with the public key
+
+###### `GroupAssignment`
+**Link**: [GroupAssignment.java](signature-lib%2FGroupAssignment.java)
+
+**Description**: An enum to clarify which group public keys and signatures are in, for a given SignatureSchema
+
+
+##### Implementation Note
+The serialization of the elements in this module adds a byte to represent the combination of Curve type and group assignment.
+
+##### Examples
```

Comment:
In the examples, rather than shorthand variable names, perhaps name each variable to describe what it is in an example.
For example:
instead of `s.verify(pbk, message)`  write `signatureToVerify.verify(pairingPublicKey, messageSigned)`.
That would be much clearer for the reader, trying to follow through multiple examples in series.

Reply:
Makes sense, thanks for pointing it out.